packages/flutter_tools/lib/src/build_runner/devfs_web.dart

```diff
@@ -337,6 +352,12 @@ class WebAssetServer implements AssetReader {
       file = globals.fs.file(webPath);
     }
 
+    if (!file.existsSync() && enableIndexRewrite) {
```

Comment:
Oh I see, this one too

Reply:
This handles the case if index.html is missing we return a 404.